File: 3168-minimum-number-of-chairs-in-a-waiting-room/3168-minimum-number-of-chairs-in-a-waiting-room.cpp
class Solution {
public:
    int minimumChairs(string s) {
        int chair = 0;
        int available = 0;
        for (char c : s) {
            if (c == 'E') {
                available--;
                chair = min(chair, available);
            } else {
                available++;
                chair = min(chair, available);
            }
        }
        return chair * (-1);
    }
};

File: 3168-minimum-number-of-chairs-in-a-waiting-room/3168-minimum-number-of-chairs-in-a-waiting-room.py
class Solution:
    def minimumChairs(self, s: str) -> int:
        chair = 0
        available = 0
        for c in s:
            if c == 'E':
                available -= 1
                chair = min(chair, available)
            else:
                available += 1
                chair = min(chair, available)
        return chair * (-1)